sandy/fy.py

```diff
@@ -357,6 +357,115 @@ def apply_qmatrix(self, zam, energy, decay_data):
+
+        """
+        S = _gls_setup(decay_data, kind, index_columns=None)
+        # Filter FY data:
+        fy_data = self.filter_by('ZAM', zam)\
+                      .filter_by("E", e).data\
+                      .set_index('ZAP')[['MT', 'DFY']]
+        # Divide the data type:
+        fy_type = fy_data.groupby('MT')
+        DIFY, DCFY = [fy_type.get_group(x)['DFY'] for x in fy_type.groups]
+        # Create the Cov object:
+        DIFY = sandy.CategoryCov.from_var(DIFY)
+        if Vy is None:
+            Vy = sandy.CategoryCov.from_var(DCFY).data
+        # Fix the index:
+        S = S.loc[DIFY.index, DIFY.index].fillna(0)
+        return DIFY.gls_update(S, Vy, threshold=threshold).data
+
+    def gls_update(self, zam, e, decay_data, xp, Vy=None, kind='cumulative',
+                   threshold=None, index_columns=None):
+        """
+        Update IFY for a given zam, energy, decay_data and new CFY information.
+
+        Parameters
+        ----------
+        zam : `int`
+            ZAM number of the material to which calculations are to be
+            applied.
+        e : `float`
+            Energy to which calculations are to be applied.
+        decay_data : `sandy.DecayData`
+            Radioactive nuclide data for several isotopes.
+        xp : 1D iterable
+            New CFY information vector.
+        Vy : 2D iterable, optional
+            Extra Covariance matrix (MXM). The default is None, so the method
+            uses the variance of the CFY for a given zam and e.
+        kind : `str`, optional
+            Keyword for obtaining prior sensitivity. The
+            default is 'cumulative'.
+        threshold : `int`, optional
+            Optional argument to avoid numerical fluctuations or
+            values so small that they do not have to be taken into
+            account. The default is None.
+        index_columns : `str`, optional
+            Keyword to determine if the sensitivity is the transposed one.
+            The default is None.
+
+        Returns
+        -------
+        `FY`
+            IFY updated with GLS for a given zam, energy, decay_data and
+            new CFY information.
+
+        """
+        data = self.data
+        S = _gls_setup(decay_data, kind, index_columns)
+        # Filter FY data:
+        fy_data = self.filter_by('ZAM', zam)\
+                      .filter_by("E", e).data\
+                      .set_index('ZAP')[['MT', 'FY', 'DFY']]
+        # Divide the data:
+        fy_type = fy_data.groupby('MT')
+        IFY, CFY = [fy_type.get_group(x)['FY'] for x in fy_type.groups]
+        DIFY, DCFY = [fy_type.get_group(x)['DFY'] for x in fy_type.groups]
+        # Put the data in a appropiate format:
+        DIFY = sandy.CategoryCov.from_var(DIFY).data
+        if Vy is None:
+            Vy = sandy.CategoryCov.from_var(DCFY).data
+        # Fix the index:
+        S = S.loc[IFY.index, IFY.index].fillna(0)
```

Comment:
this is not True, once again, sensitivities might have different indices (you will see it with chain yields, where indices are not nuclides but mass numbers)